Paintroid/src/main/java/org/catrobat/paintroid/tools/drawable/DrawableFactory.kt

```diff
@@ -16,16 +16,15 @@
  * You should have received a copy of the GNU Affero General Public License
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
+package org.catrobat.paintroid.tools.drawable
 
-package org.catrobat.paintroid.tools.drawable;
-
-import android.graphics.Canvas;
-import android.graphics.Paint;
-import android.graphics.RectF;
-
-public class RectangleDrawable implements ShapeDrawable {
-	@Override
-	public void draw(Canvas canvas, RectF shapeRect, Paint drawPaint) {
-		canvas.drawRect(shapeRect, drawPaint);
-	}
+class DrawableFactory {
+    fun createDrawable(shape: DrawableShape): ShapeDrawable {
+        return when (shape) {
+            DrawableShape.RECTANGLE -> RectangleDrawable()
+            DrawableShape.OVAL -> OvalDrawable()
+            DrawableShape.HEART -> HeartDrawable()
+            DrawableShape.STAR -> StarDrawable()
```

Comment:
Nice 👍 
Obviously, the Kotlin compiler recognises that DrawableShape can only be those four, hence no IllegalArgumentException 😄 
Unfortunately, Github compares the wrong files (not your fault)

Reply:
yeah. Thanks! 😄